.github/workflows/submitty_ci.yml

```diff
@@ -133,58 +133,106 @@ jobs:
-          with:
-            files: autograder/coverage.xml
-            flags: autograder
         - name: Upload submitty python utils Coverage
           uses: codecov/codecov-action@v1
           with:
             files: python_submitty_utils/coverage.xml
             flags: python_submitty_utils
+
+    python-unit-migration:
+      runs-on: ubuntu-18.04
+      steps:
+        - uses: actions/checkout@v2
+        - uses: actions/setup-python@v2
+          with:
+            python-version: '3.6'
+        - name: Cache Pip
+          uses: actions/cache@v2
+          with:
+            path: ~/.cache/pip
+            key: ${{ runner.os }}-pip-${{ hashFiles('**/migration_requirements.txt') }}
+            restore-keys: |
+                ${{ runner.os }}-pip-
+        - name: Install python libraries
+          working-directory: migration
+          run : |
+            python3 -m pip install --upgrade pip setuptools wheel
+            python3 -m pip install coverage
+            python3 -m pip install -r migration_requirements.txt
+        - name: Run migrator python unit tests
+          working-directory: migration
+          run : |
+            coverage run -m unittest discover
+            coverage xml
         - name: Upload migrator Coverage
           uses: codecov/codecov-action@v1
           with:
             files: migration/coverage.xml
             flags: migrator
 
-
+    python-unit-auto-grader:
+      runs-on: ubuntu-18.04
+      # needs: [python-unit-submitty-utils] Not that necessary
+      steps:
+        - uses: actions/checkout@v2
+        - uses: actions/setup-python@v2
+          with:
+            python-version: '3.6'
+        - name: Cache Pip
+          uses: actions/cache@v2
+          with:
+            path: ~/.cache/pip
+            key: ${{ runner.os }}-pip-${{ hashFiles('**/autograder_requirements.txt') }}
+            restore-keys: |
+              ${{ runner.os }}-pip-
+        - name: Install python libraries
+          working-directory: autograder
+          run : |
+            python3 -m pip install --upgrade pip setuptools wheel
+            python3 -m pip install coverage
+            python3 -m pip install ../python_submitty_utils/
+            python3 -m pip install -r autograder_requirements.txt
+        - name: Run autograder python unit tests
+          working-directory: autograder
+          run : |
+            coverage run -m unittest discover
+            coverage xml
+        - name: Upload autograder Coverage
+          uses: codecov/codecov-action@v1
+          with:
+            files: autograder/coverage.xml
+            flags: autograder
 
     e2e:
-        needs: [eslint, php-lint-static, php-unit, python-lint, python-unit]
+        needs: [
+          eslint,
+          php-lint-static,
+          php-unit,
+          python-lint,
+          python-unit-migration,
+          python-unit-auto-grader,
+          python-unit-submitty-utils
+        ]
```

Comment:
```suggestion
        needs:
          - eslint
          - php-lint-static
          - php-unit
          - python-lint
          - python-unit-migration
          - python-unit-auto-grader
          - python-unit-submitty-utils
```

Reply:
Wow, didn't know this before! This is neat! Thanks 👍